ciao-cli/main.go

```diff
@@ -855,127 +295,96 @@ func checkCompulsoryOptions() {
 		fatal += "Missing required Ciao controller URL\n"
 	}
 
-	if *allInstances == true && *instance != "" {
-		fatal += "All instances or one single instance ?\n"
-	}
-
 	if fatal != "" {
 		fatalf(fatal)
 	}
 }
 
-func cliList() {
-	if *listInstances == true {
-		listAllInstances(*tenantID, "", *instanceMarker, *instanceOffset, *listLength)
-	}
-
-	if *listWlInstances == true {
-		listAllInstances("", *workload, *instanceMarker, *instanceOffset, *listLength)
-	}
-
-	if *listCNInstances == true {
-		listNodeInstances(*computeNode)
-	}
-
-	if *listQuotas == true {
-		listTenantQuotas(*tenantID)
-	}
-
-	if *listResources == true {
-		listTenantResources(*tenantID)
-	}
-
-	if *listWorkloads == true {
-		listTenantWorkloads(*tenantID)
-	}
-
-	if *listComputeNodes == true {
-		listAllComputeNodes()
-	}
+func main() {
+	var err error
 
-	if *listCNCIs == true {
-		listAllCNCIs()
-	}
+	flag.Usage = usage
+	flag.Parse()
 
-	if *listLabels == true {
-		listAllLabels()
-	}
+	getCiaoEnvVariables()
+	checkCompulsoryOptions()
 
-	if *listEvents == true || *listAllEvents == true {
-		listClusterEvents(*tenantID, *listAllEvents)
+	/* If we're missing the tenant name let's try to fetch one */
+	if *tenantName == "" {
+		*tenantName, *tenantID, err = getTenant(*identityUser, *identityPassword, *tenantID)
+		if err != nil {
+			fatalf(err.Error())
+		}
+		warningf("Unspecified scope, using (%s, %s)", *tenantName, *tenantID)
 	}
-}
 
-func cliDump() {
-	if *clusterStatus == true {
-		dumpClusterStatus()
+	scopedToken, *tenantID, _, err = getScopedToken(*identityUser, *identityPassword, *tenantName)
+	if err != nil {
+		fatalf(err.Error())
 	}
 
-	if *dumpCNCI == true {
-		dumpCNCIDetails(*cnci)
+	// Print usage if no arguments are given
+	args := flag.Args()
+	if len(args) < 1 {
+		usage()
 	}
 
-	if *dumpLabel != "" {
-		dumpTraceData(*dumpLabel)
+	// Find item in cmdline args
```

Comment:
@albertomurillo I'll take that back. We want to support the following format:
`ciao-cli [generic options] [command] [subcommand] [command specific options]`

So I think your original options handling implementation is correct.